sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/LocalDocumentStorage.java

```diff
@@ -418,3 +425,34 @@ void updatePendingOperation(LocalDocument operation) {
         return new DocumentWrapper<>(new DataException("Document was not found in the cache."));
     }
 }
+
+class LocalDocumentStorageDatabaseListener implements DatabaseManager.Listener {
+
+    private String userTable;
+
+    LocalDocumentStorageDatabaseListener(String userTable) {
+        this.userTable = userTable;
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+    }
+
+    @Override
+    public boolean onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        /*
+            Between versions 1 and 2 we need to drop both user and readonly tables
+            because we discovered we had been doing inserts instead of upserts into the cache in the original SDK release.
+            Dropping the table will allow to clean up what essentially are duplicate rows
+            (the same partition and document ID, but different `oid`).
+         */
+        if (oldVersion == 1 && newVersion == 2 && userTable != null) {
```

Comment:
why testing newVersion == 2? If later you upgrade from 1 to 3 directly you also need to execute that. We just need to check oldVersion is 1

Reply:
Also why the fix does not apply to app documents?

Reply:
App documents will get deleted automatically by the `DatabaseManager` (line 123) as the comment points out.